### PR TITLE
pfs: adds WorkerEnv, WorkerConfig, and Worker

### PR DESCRIPTION
**Pull request by brendoncarroll**
Separates PFS worker code into its own type. The PFS worker will only run in pachws in production, but is also setup by real_env during tests.

WorkerEnv only contains Etcd, DB, and ObjClient.  It doesn't need the rest of the PFS DAG, the listeners, or the Auth Service.  In order to break the dependency on the Auth Service, the GetFileURL task had to be changed to deal with Filesets, rather than Commits.  As a general rule, Filesets should be passed around to refer to data, not Commits.
Breaking the dependency on the Auth Service, transitively remove the dependency on Enterprise and License.

### CODE REVIEW

File: src/internal/pachd/builder.go [hunk 1]
```diff
@@ -408,6 +408,22 @@ func (b *builder) initPachwController(ctx context.Context) error {
...
+		return err
+	}
+	config := pfs_server.WorkerConfig{
+		Storage: b.env.Config().StorageConfiguration,
+	}
+	w, err := pfs_server.NewWorker(*env, config)
+	if err != nil {
+		return err
+	}
+	go w.Run(ctx) //nolint:errcheck
```
Comment (albscui) [inline src/internal/pachd/builder.go:423]: Should we log the error like how you did it in testpachd?
```
go func() {
		if err := w.Run(ctx); err != nil {
			log.Error(ctx, "from worker", zap.Error(err))
		}
	}()
```

**Review by albscui — APPROVED** [review]
LGTM. Just left a few nits.

File: src/server/pfs/server/worker_url.go [hunk 2]
```diff
@@ -120,18 +116,28 @@ func (d *driver) processGetFileURLTask(ctx context.Context, task *GetFileURLTask
...
 				return errors.EnsureStack(err)
 			}
-			defer errors.Close(&retErr, w, "close writer for bucket %s", url.Bucket)
-			return errors.EnsureStack(file.Content(ctx, w))
-		})
-		return errors.EnsureStack(err)
+			if err := f.Content(ctx, w); err != nil {
+				return err
+			}
+			return errors.EnsureStack(w.Close())
```
[the remark this answers is not in the document]
Comment (brendoncarroll) [reply src/server/pfs/server/worker_url.go:139]: I was looking at this, and I don't think we want to do that.  Closing actually completes the operation, which in the case of an error would leave an incomplete file.  I believe the context passed to the constructor manages cleanup in the failure case.

File: src/internal/pachd/builder.go (hunk 1 above)
Comment (brendoncarroll) [reply src/internal/pachd/builder.go:423]: Yeah you're probably right, will do.

File: src/server/pfs/server/worker.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		ObjectStore: env.ObjClient,
+		DB:          env.DB,
+	}, config.Storage)
+	if err != nil {
+		return nil, err
+	}
+	return &Worker{
+		env:    env,
+		config: config,
+
```
Comment (FahadBSyed) [inline src/server/pfs/server/worker.go:45]: [nit] It seems like in the PFS codebase we prefer not adding empty newlines, not sure if its worth standardizing on this, but I figured I should call that out.

File: src/server/pfs/server/worker.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		storage: ss,
+	}, nil
+}
+
+func (w *Worker) Run(ctx context.Context) error {
+	eg, ctx := errgroup.WithContext(ctx)
+	log.Info(ctx, "started worker")
+	defer log.Info(ctx, "exited worker")
+	// compactionWorker
```
Comment (FahadBSyed) [inline src/server/pfs/server/worker.go:54]: [nit] You probably don't need these comments, these calls to eg.Go() seem straightforward enough.

File: src/server/pfs/server/worker_url.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (FahadBSyed) [reply src/server/pfs/server/worker_url.go:139]: I think it doesn't have to be deferred because its the final statement in the function.

**Review by FahadBSyed — APPROVED** [review]
left some non-blocking comments, other than that looks good to me.

**Pull request merged**